src/aspire/source/__init__.py

```diff
@@ -346,8 +345,8 @@ def downsample(self, L):
         self.generation_pipeline.add_xform(Downsample(resolution=L))
 
         ds_factor = self.L / L
-        for f in set(self.filters):
-            f.scale(ds_factor)
+        for f_ind in range(len(self.unique_filters)):
```

Comment:
You don't need `f_ind` here. The old way would have worked just fine.

Reply:
Removed in new version.